File: ninja-fe/src/model/api/conversation/ConversationCriteria.ts
export interface ConversationCriteria {
  globalSearch?: string;
  minMessageCount?: number;
  maxMessageCount?: number;
  minCreatedAt?: string;
  maxCreatedAt?: string;
  summary?: string;
  userFullName?: string;
  deleted?: boolean;

  getUrl(): string;
}

File: samurai-be/src/main/java/pl/ochnios/samurai/model/dtos/conversation/ConversationCriteria.java
package pl.ochnios.samurai.model.dtos.conversation;

import io.swagger.v3.oas.annotations.media.Schema;
import java.time.Instant;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@Schema(description = "${docs.dto.conversationCriteria}")
public class ConversationCriteria {

    @Schema(description = "${docs.dto.conversationCriteria.globalSearch}")
    private String globalSearch;

    private Integer minMessageCount;
    private Integer maxMessageCount;
    private Instant minCreatedAt;
    private Instant maxCreatedAt;
    private String summary;
    private String userFullName;
    private Boolean deleted;
}
